stardog-userrole-operator/test/deployment_test.go

```diff
@@ -0,0 +1,110 @@
+package test
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/stretchr/testify/assert"
+	appv1 "k8s.io/api/apps/v1"
+)
+
+var (
+	tplDeployment = []string{"templates/deployment.yaml"}
+)
+
+func Test_Deployment_ShouldRender_DefaultServiceAccount(t *testing.T) {
+	want := releaseName + "-stardog-userrole-operator"
+	options := &helm.Options{}
+
+	got := renderDeployment(t, options, false)
+	serviceName := got.Spec.Template.Spec.ServiceAccountName
+	assert.Equal(t, want, serviceName, "Deployment does not render configured serviceName")
+}
+
+func Test_Deployment_ShouldRender_CustomServiceAccount(t *testing.T) {
+	want := "test"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"serviceAccount.name": want,
+		},
+	}
+
+	got := renderDeployment(t, options, false)
+
+	serviceName := got.Spec.Template.Spec.ServiceAccountName
+	assert.Equal(t, want, serviceName, "Deployment does not render configured serviceName")
+}
+
+func Test_Deployment_ShouldRender_Resources(t *testing.T) {
+	want := "1Gi"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"resources.limits.memory": want,
+		},
+	}
+
+	got := renderDeployment(t, options, false)
+	resources := got.Spec.Template.Spec.Containers[0].Resources
+	assert.Equal(t, want, resources.Limits.Memory().String(), "Deployment does not render configured memory limit")
+}
+
+func Test_Deployment_ShouldRender_Labels(t *testing.T) {
+	options := &helm.Options{}
+
+	got := renderDeployment(t, options, false)
+
+	selector := got.Spec.Selector.MatchLabels
+	matchLabels := got.Spec.Template.Labels
+	assert.Equal(t, selector, matchLabels, "Deployment does not render matching labels")
+}
+
+func Test_Deployment_ShouldRender_ImagePullSecrets(t *testing.T) {
+	expectedImageSecret := "secret"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"imagePullSecrets[0].name": expectedImageSecret,
+		},
+	}
+
+	got := renderDeployment(t, options, false)
+
+	assert.Equal(t, expectedImageSecret, got.Spec.Template.Spec.ImagePullSecrets[0].Name, "Deployment does not render image pull secrets")
+}
+
+func Test_Deployment_ShouldRender_CorrectImage(t *testing.T) {
+	expectedImage := "docker.io/vshn/stardog-userrole-operator:0.0.1"
+	options := &helm.Options{}
+
+	got := renderDeployment(t, options, false)
+
+	assert.Equal(t, expectedImage, got.Spec.Template.Spec.Containers[0].Image, "Deployment does not render image location")
+}
+
+func Test_Deployment_WhenProxyEnabled_ThenEnableRBACContainer(t *testing.T) {
+	expectedBaseImage := "docker.io/vshn/stardog-userrole-operator:0.0.1"
+	expectedProxyImage := "gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"metrics.proxy.enabled": "true",
+		},
+	}
+
+	got := renderDeployment(t, options, false)
+
+	assert.Equal(t, 2, len(got.Spec.Template.Spec.Containers), "Deployment does not have correct nr of containers")
```

Comment:
Would change the wording:

```suggestion
	assert.Equal(t, 2, len(got.Spec.Template.Spec.Containers), "Deployment does not have the correct amount of containers")
```